client/my-sites/checkout/checkout/web-payment-box.jsx

```diff
@@ -0,0 +1,517 @@
+				},
+			},
+		];
+		const paymentDetails = {
+			total: {
+				label: translate( 'Total' ),
+				amount: {
+					currency: cart.currency,
+					value: cart.total_cost,
+				},
+			},
+			displayItems: cart.products.map( product => {
+				return {
+					label: product.product_name,
+					amount: {
+						currency: product.currency,
+						value: product.cost,
+					},
+				};
+			} ),
+		};
+
+		return new PaymentRequest( supportedPaymentMethods, paymentDetails, PAYMENT_REQUEST_OPTIONS );
+	};
+
+	/**
+	 * @param {string} paymentMethod  Payment method.
+	 * @param {DOMEvent} event        Button even.
+	 */
+	submit = ( paymentMethod, event ) => {
+		event.persist();
+		event.preventDefault();
+
+		switch ( paymentMethod ) {
+			case WEB_PAYMENT_APPLE_PAY_METHOD:
+				{
+					const paymentRequest = this.getPaymentRequestForApplePay();
+
+					try {
+						paymentRequest
+							.show()
+							.then( paymentResponse => {
+								const { payerName, details } = paymentResponse;
+								const { token } = details;
+
+								if ( 'EC_v1' !== token.paymentData.version ) {
+									return; // Not supported yet.
+								}
+
+								this.props.transaction.payment = newCardPayment( {
+									tokenized_payment_data: token.paymentData,
+									name: payerName,
+									country: this.state.country,
+									'postal-code': this.state.postalCode,
+									card_brand: token.paymentMethod.network,
+								} );
+
+								setPayment( this.props.transaction.payment );
+
+								this.props.onSubmit( event );
+
+								paymentResponse.complete( 'success' );
+							} )
+							.catch( error => {
+								debug( 'show error' );
+								debug( error );
+							} );
+					} catch ( e ) {
+						debug( 'show catch' );
+						debug( e );
+					}
+				}
+				break;
+
+			case WEB_PAYMENT_BASIC_CARD_METHOD:
+				{
+					const paymentRequest = this.getPaymentRequestForBasicCard();
+
+					try {
+						paymentRequest
+							.show()
+							.then( paymentResponse => {
+								const { details } = paymentResponse;
+								const { billingAddress } = details;
+
+								const cardRawDetails = {
+									number: details.cardNumber,
+									cvv: details.cardSecurityCode,
+									'expiration-date': details.expiryMonth + '/' + details.expiryYear.substr( 2 ),
+									name: details.cardholderName,
+									country: billingAddress.country,
+									'postal-code': billingAddress.postalCode,
+								};
+
+								this.props.transaction.payment = newCardPayment( cardRawDetails );
+
+								paymentResponse.complete();
+
+								return this.props.transaction.payment;
+							} )
+							.then( transactionPayment => {
+								setPayment( transactionPayment );
+								this.props.onSubmit( event );
+							} )
+							.catch( error => {
+								debug( error );
+							} );
+					} catch ( e ) {
+						debug( e );
+					}
+				}
+				break;
+
+			default:
+				debug( 'Unknown or unhandled payment method `' + paymentMethod + '`.' );
```

Comment:
This is fine but you could use a template literal instead of concatenation (https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Template_literals) or you could use a comma and just output `paymentMethod` directly.

Reply:
Done.